app/helpers/bookmarks_helper.rb

```diff
@@ -84,4 +84,52 @@ def get_count_for_bookmark_blurb(bookmarkable)
     content_tag(:span, link, class: "count")
   end
 
+  def bookmarker_id_for_css_classes(bookmark)
+    return if bookmark.nil?
+
+    "user-#{bookmark.pseud.user_id}"
+  end
+
+  # Array of unique creator and bookmarker ids, formatted user-123, user-126.
+  # If the user has bookmarked their own work, we don't need their id twice.
+  def user_ids_for_bookmark_blurb(bookmark)
+    user_ids = creator_ids_for_css_classes(bookmark.bookmarkable)
+    user_ids << bookmarker_id_for_css_classes(bookmark)
+    user_ids.uniq
+  end
+
+  # The bookmark blurb contains the bookmarkable and a single user's bookmark.
+  # If we cache this, it probably only needs to use the creation cache key.
```

Comment:
Doesn't it also need to use the bookmark's cache key, because it also includes the bookmarker's user ID?

I think it probably should be cached, at least until we get a bunch of N+1 errors fixed up. And that's probably not going to happen before the next release.

Reply:
I was going to say yes, but... does it? The user ID for the bookmarker will never change. You can't transfer a bookmark to a different account, nor can you make it anonymous... you can only delete it.

Reply:
It can't change, that's true, but the contents of the cache includes the bookmarker ID, so the bookmark has to be used somewhere in the cache key. You could get away with using `bookmark.id` instead of the full `bookmark.cache_key`, though.